src/Yarhl/IO/IStream.cs

```diff
@@ -26,6 +26,11 @@ namespace Yarhl.IO
     /// </summary>
     public interface IStream : IDisposable
     {
+        /// <summary>
+        /// Gets or sets the lock object of this stream (and all its substreams).
+        /// </summary>
+        object LockObj { get; set; }
```

Comment:
I think for the interface it makes sense to just require the _getter_ (just in case some implementations doesn't want to allow setting from external). What do you think?

Reply:
I have to think about it, but it might be useful if we want to share the same lock object between unrelated Streams.